src/main/scala/firrtl/transforms/CheckCombLoops.scala

```diff
@@ -199,8 +199,12 @@ class CheckCombLoops extends Transform {
       val moduleGraph = DiGraph(internalDeps)
       moduleGraphs(m.name) = moduleGraph
       simplifiedModuleGraphs(m.name) = moduleGraphs(m.name).simplify((m.ports map { p => LogicNode(p.name) }).toSet)
-      for (scc <- moduleGraphs(m.name).findSCCs.filter(_.length > 1)) {
-        val sccSubgraph = moduleGraphs(m.name).subgraph(scc.toSet)
+      // Find combinational nodes with self-edges; this is *NOT* the same as length-1 SCCs!
+      for (unitLoopNode <- moduleGraph.getVertices.filter(v => moduleGraph.getEdges(v).contains(v))) {
+        errors.append(new CombLoopException(m.info, m.name, Seq(unitLoopNode.name)))
+      }
```

Comment:
Are we sure that Scala or the JVM is going to CSE the inner `moduleGraph.getEdges(v).contains(v)`? If it doesn't this would be very slow.

Reply:
Not sure I follow

Reply:
Nevermind, I was assuming `.getEdges` iterates on all of the edges like `getVertices` iterates on all of the vertices does but it does not 😇

Reply:
Yeah, I was confused, since this is **|V|** times two hash table lookups. It's so much faster than the rest of the pass :)

Reply:
Yeah I was worried it was **|V| * |E|**... which I guess would still have been faster than the rest of the pass 🤷‍♀️

Reply:
No, the pass runs in **|E|** lol

Reply:
Oh word 💯